feat(ui): Storybook: Add Updates page & move Overview page

vuluongj20 opened the pull request:
**Added an Updates page** to document the latest changes.

<img width="913" alt="Screen Shot 2021-09-22 at 6 39 50 PM" src="https://user-images.githubusercontent.com/44172267/134443417-f6faee23-9593-4010-b032-2c6188304ef0.png">

---

**Moved the Overview page** out to the sidebar root to live next to the Updates page, and added links to the bottom.

<img width="227" alt="Screen Shot 2021-09-22 at 6 45 31 PM" src="https://user-images.githubusercontent.com/44172267/134443534-80572e9f-37ee-415d-beb4-bb42fcc115cb.png">

vuluongj20 committed:
Storybook: Add Updates page & move Overview page

 - Added an Updates page to document the latest changes to the design and component system
 - Moved the Overview page out to the sidebar root, and added links at the bottom

vuluongj20 commented on .storybook/main.ts:
```diff
@@ -10,7 +10,7 @@ import babelConfig from '../babel.config';
 
 const config: StorybookConfig = {
-  stories: ['../docs-ui/stories/*/*.stories.*'],
+  stories: ['../docs-ui/stories/**/*.stories.*'],
```
To get .stories pages at all depths

priscilawebdev committed:
feat(alerts): Add project incident aler to proj/issue stream page - NATIVE-216 & NATIVE-217 (#28726)

armenzg committed:
feat(dev): Install postgreqsl on Apple M1 to build psycopg2-binary (#28607)

Create a new target which installs different libraries via brew depending on the context.

For instance, on Apple M1 machines it should installed a couple more libraries.
This helps to install the psycopg2-binary package since there's no wheel for it.
We also upgrade the version of it in order to avoid needing to set some CPPFLAGS & LDFLAGS.

In the future, the bootstrap script and documentation should be using this target.

Fixes #28606

jjbayer committed:
feat(release-archives): Add spans to fetch_release_archive_for_url (#28790)

Add more instrumentation to function that triggers soft
timeouts.

https://getsentry.atlassian.net/browse/INGEST-116

Zylphrex committed:
fix(discover): Should reload when payload changes while loading (#28653)

If the eventView changes while the request is in-flight/loading, we should
cancel the in-flight request and fetch again. This is problematic in cases where
the request takes a little longer to load and the user changes the query while
it is still in-flight. If a reload is not triggered at this time, the results
are actually incorrect since it's showing the results from the previous (stale)
in-flight request.

oioki committed:
feat: Add similarity.index queue (#28793)

markstory committed:
ref(deletion) Remove unused deletion tasks (#28463)

With all of these async deletions being done with ScheduledDeletion we
no longer need to have these tasks and their tests around.

Zylphrex committed:
feat(snql): Use aliased expression instead of function hack (#28740)

Given the `AliasedExpression` from the sdk, we can now remove the hack we were
previously using with noop functions to get aliases.

getsentry-bot committed:
Revert "ref(deletion) Remove unused deletion tasks (#28463)"

This reverts commit c18a6ac9e2e0128fb29f2142fbf660b4e3d59449.

Co-authored-by: mark via Slack <mark@sentry.io>

mgaeta committed:
fix(webhook): Weird JSON array (#28545)

wmak committed:
fix(features): Skip the entity handler after batch checking (#28611)

* fix(features): Skip the entity handler after batch checking

- This adds the ability to skip the entity handler entirely when calling
  features.has
- This is becaue when there are a lot of non entity features to check
  we're actually spending time checking that the feature is indeed not
  an entity feature

wmak committed:
fix(discover): Handle field aliases with others correctly (#28751)

* fix(discover): Handle field aliases with others correctly

- The Other query doesn't include any selected columns which means that
  using the FIELD_ALIAS's alias won't have a corresponding column
- This fixes that issue by pulling the column from the selected columns
  and uses it in the conditions

scttcper committed:
fix(workflow): Add alpha badge to project reports, add title (#28770)

wmak committed:
fix(discover): Fix a colour mismatch issue with other results (#28736)

- When other is included in top events, colours wouldn't match because
  the count of series would mismatch (ie. 6 from stats results, but 5
  for the table)
  - This increases the total count by one if the others flag is on +
    theres more than top-n results

matejminar committed:
fix(ui): Remove % from failure rate alert threshold (#28800)

mikellykels committed:
fix(semver): Handle NOT operator for releases list (#28713)

* fix(semver): Handle NOT operator for releases list

Handle NOT operator when filtering through the Releases List.

FIXES WOR-1284

* refactor negation code, add more tests

* remove unused code

RyanSkonnord committed:
test: Fix setup in test_flow_authenticated_without_verified_without_password (#28784)

Change create_user so that calling it with password="" sets up the user
in a state where it does not have a usable password, as the test case
intended. Previously, any falsy value would give it a default password.

Update the assertion to match correct behavior.

edwardgou-sentry committed:
feat(discover): added exclusive selection when clicking on option checkbox row

Updates the Y-Axis dropdown in discover results chart to exclusively select options when clicking on the row and multi select options when clicking on the checkbox

RyanSkonnord committed:
feat(auth): Stub for automatically migrating to new SSO provider (#28621)

When a user supplies unrecognized SSO credentials associated with an
email address for an existing user profile, offer them a chance to link
their identity to that profile instead of automatically provisioning a
new profile.

Introduce a feature flag. The feature is incomplete and will hit
unimplemented code if the flag is set.

shruthilayaj committed:
chore(chart-unfurl): Revert Project permissions warning (#28802)

We now prompt users to link their Slack identity to
Sentry which we use to honour project permissions.
This warning no longer applies. Revert of #28371

armenzg committed:
feat(dev): Upgrade cryptography on Apple's M1 machines (#28769)

The newer version includes an arm64 wheel.

scttcper committed:
feat(workflow): Save selected insights team in localstorage (#28774)

k-fish committed:
feat(perf): Add page error context (#28805)

Splitting off more pieces from the perf. landing v3. This is just a page error context which can be used without drilling setError everywhere. Added a simple test so it's not just implicitly tested.

leeandher committed:
fix(alert-rules): Set the correct default value for new SelectControl (#28804)

RyanSkonnord committed:
fix(test): Preserve default feature flag values when patching in tests (#28686)

Change the Feature object, which is used to patch feature flag values in
unit tests, so that it only affects the named flags and preserves
default behavior for the remaining flags. Previously, patching with it
would cause all flags to default to false even if they would default to
true ordinarily.

Change affected tests to declare certain flags explicitly as false,
where the test would fail otherwise.

davidenwang committed:
ref(select): Choices -> options integrationListDirectory (#28756)

davidenwang committed:
ref(select): Choices -> options monitorForm (#28755)

* choices to options

* add type for intervals

wedamija committed:
feat(comparison_alerts): Translate delta percents into comparison percents in the api layer. (#28787)

The frontend represents percentages as delta change percents. For example:
 - 30% increase
 - 40% decrease.

This makes sense from a user point of view, but doesn't fit as well into how our system handles
thresholds. For example, with decrease, if we had threshold like:

- critical at 40%
- resolve at 30%

This would fail our validation, since resolve thresholds on below alerts need to be greater than the
critical/warning thresholds. It also makes comparisons in the subscription processor more difficult.

To work around this, we convert these delta percents into comparison percents. This means that:

 - 30% increase -> 130% comparison percent
 - 40% decrease -> 60% comparison percent.

This fits well into our validation, since now if we have a decrease % alert like

- critical at 40% (60% comparison)
- resolve at 30% (70% comparison)

The resolve threshold is now higher than the critical, and validation will fire.

Handling this complexity at the api layer seems to work the best, so that internally we can assume
that all of these threshold are comparison percents.

I also considered making increase deltas positive and decrease deltas negative, but I felt like this
reads weirdly in terms of our data schema (we'd have a below trigger that looks like < -30%, if you
just looked at the data).

wedamija committed:
fix(semver): Fix failed semver search in issue search/discover when negative search returns too many results (#28813)

This fixes a bug when using negative search on semver. When we filter, we fetch up to 1000 releases.
For any org with over 1000 releases, a negative search fall back to flipping the search around. This
fails since the operator negation map doesn't define `!=` and `NOT IN`.

priscilawebdev committed:
fix(dynamic-sampling): Fix sampling rule not remembering  trace flag - INGEST-411 (#28795)

priscilawebdev committed:
fix(dynamic-sampling): Fix minor bugs on mobile (#28820)

jjbayer committed:
feat(release-archives): Max. archive size option for disk caching (#28821)

Add an option which limits the caching of release archives to a certain
size.

This prepares for #28818.

https://getsentry.atlassian.net/browse/INGEST-116

jjbayer committed:
fix(release-archives): Do not attempt to cache large ZIP files  (#28818)

For very large release archives, loading the entire zip file may exceed
the time limit for the current processing task. The only use case for
loading an entire zip file at once is caching - in order to extract a
single entry from a zip file, only the central directory and the file
entry itself are required.

This PR prevents loading large zip files for both disk and memory
caching.

https://getsentry.atlassian.net/browse/INGEST-116

markstory committed:
ref(ui) Convert newsletter form to new form components (#28785)

Minor layout changes from using the new components.

RaduW committed:
ref(releasehealth): Implement check_has_health_data on metrics backend [INGEST-249] (#28729)

 check_has_health_data implemented for metrics

beezz committed:
fix(python3): Remove itertools.ifilter use (#28822)

k-fish committed:
feat(perf): Add tables to all tabs in landing v3 (#28806)

This just adds the tables with their columns for all the landing v3 tables. The mobile tab still needs the react native check, but that can be addressed when I get to filling the rest of that tab out. Also adds basic tests for all the tabs.

shruthilayaj committed:
feat(dashboards): Add top n events support (#28734)

Add top n events type widget to dashboards.
This allows user to add multiple fields for the
top n groupby. Restricting the column edit section
to not have any aggregate options.

leeandher committed:
fix(alert-rules): Force the RuleNode select values to be strings (#28828)

shruthilayaj committed:
chore(analytics): Add some analytics around chart unfurls (#28807)

Track how many users click on the "Link" button
and how many links are posted and how many
of them are unfurled.

joshuarli committed:
fix: safe_urlencode (#28812)

wedamija committed:
fix(cdc_search): Fix bug with groups that have no users when sorting by unique users (#28618)

When sorting by unique users, if no events within the group have users then this can return None for
the score, which causes problems in the paginator. Just default any `None` scores to 0.

Fixes SENTRY-S68
Fixes SENTRY-S61

vuluongj20 committed:
feat(ui): Storybook DocsLinks (#28786)

* Update DocsLinks UI

- The title is now smaller, more in line with the rest of the documentation UI
- There is now an optional `desc` prop for rendering a small line of text below the title, useful for longer descriptions of the linked page

* Disable ESLint error

For some reason ESLint thinks that there is no default export in `@storybook/addon-links/react`. There is, and the component works normally with the current import line.

wedamija committed:
feat(comparison_alerts): Update issue alert api to support comparison values (#28661)

This allows issue alerts with comparison values to be created, and returns these new fields back
from the appropriate endpoints.

wedamija committed:
feat(comparison_alerts): Update subscription processor to perform comparison alerts (#28814)

This updates the subscription processor to be able to handle comparison alerts. The way this works
is when we receive a value, we check whether the alert rule has a `comparison_delta`. If so, then we
know this is a comparison alert. We then take the delta and subtract it from the timestamp of the
value to determine the end of the comparison period. We then query from `comparison_period_end -
time_window` to `comparison_period_end` using the same query that the subscription uses, which gives
us out comparison value. Once that's done, we just divide the value by the comparison value and
multiple by 100, which gives us our comparison percent, which can be compared to the alert rule
thresholds.

Just a note from other PRs: Comparison alerts make twice as many snuba queries as normal alerts. To
offset this we set these alerts to run every two minutes instead of every minute.

evanpurkhiser committed:
ref(js): Remove heavyweight organization (#28783)

davidenwang committed:
ref(select): Remove useState in teamSelector (#28738)

With recent changes to the <Projects> utility in #28610 we no longer have to maintain internal state here for keeping track of teams which were added to the current project.

In order to still achieve the same optimistic update that we had before with local state we now assume that if a team/option is selected then it is already part of the project's teams.

RaduW committed:
ref(releasehealth): Implement check_releases_have_health_data on metrics backend [INGEST-249]  (#28823)

Implement metrics version for check_releases_have_health_data [INGEST-249]

matejminar committed:
feat(ui): Add session chart to alert creation page (#28826)

ahmedetefy committed:
fix(semver): Backfill missed semver releases (#28798)

Migration to backfill semver columns that
were not populated incorrectly as a result of bug
that did not run the semver version parser on Release
creation

markstory committed:
feat(deletions) Move organizationintegration deletes to ScheduledDeletion (#28803)

I missed this one in my previous inventory of custom deletion tasks.
I've also expanded the Repository relations to include code mappings as
there is another foreign key there that will block deletions.

priscilawebdev committed:
ref(breadcrumbs): Rename breadcrumbs types (#28853)

untitaker committed:
ref(release_health): Rename module to be consistent with function name conventions (#28829)

shruthilayaj committed:
(discover): Remove user id from url query params (#28840)

URL query params are no longer used in discover links,
since they aren't used to determine my projects anymore.

markstory committed:
feat(ui) Add a checkbox field to new forms (#28809)

Add a single checkbox field to the new form field library. This field
will help with 'confirm' inputs like the one used during organization
creation in #28773

edwardgou-sentry committed:
fix(dashboards): Fixed add widget from discover to new dashboard not creating widget (#28827)

Fixed add widget from discover to new dashboard not creating widget

wmak committed:
ref(discover): Introduce some order to discover columns (#28832)

- This sorts fields, tags, measurements and breakdowns alphabetically
- This regroups functions slightly so that they're next to other similar
  functions (eg. avg + percentile)
  - Not sorting alphabetically cause the order of some things become
    strange (eg. p100 before p50)

MeredithAnya committed:
(sentry-metrics): Metrics indexer consumer (#28431)

* WIP(metrics): Metrics indexer consumer

* use redis for mock indexer

* add bulk_record and async producing

* make redis mock indexer separate file

* fix type errors

* add comment

* remove UseCase and updates tests

* update more tests

* clean up part I

* mini cleanup

* add basic tests

* all org_ids are ints

* missed one

* more clean up

* consumer test

* rename test file

* lil updates

* attempt to fix tests

* try dis tho

edwardgou-sentry committed:
feat(discover): Update discover to support saving multi y axis queries (#28699)

Updates discover saved queries to allow saving multiple selected y axis.

markstory committed:
ref(ui) Convert Saved Issue Search to new forms (#28839)

Update saved issue search to settings/forms. I've also replaced the
custom footer buttons with more standard size ones.

Co-authored-by: Matej Minar <matej.minar@sentry.io>
Co-authored-by: Priscila Oliveira <priscilawebdev@gmail.com>

scttcper committed:
feat(workflow): Switch project reports date range to weeks (#28847)

AniketDas-Tekky committed:
ref(api-logging): Added several more metrics (#28844)

These new metrics are: user id, organization id, auth id, remote address, and url path

AniketDas-Tekky committed:
fix(api-logging): leverage getattrs to reduce errors (#28866)

Instead of assuming things like request.access is available, it now leverages getattr to default values to None if they aren't

evanpurkhiser committed:
ref(routes): Remove SettingsProjectProvider (#28842)

edwardgou-sentry committed:
fix(discover): saved queries without y axis default to count() when comparing for changes (#28863)

Saved queries without y axis default to count() when comparing for changes. This fixes scenarios where saved queries that have undefined yAxis in the backend show change on the frontend even when no change really exists

scttcper committed:
fix(workflow): Remove color from 0% change in team user misery (#28865)

evanpurkhiser committed:
ref(js): Remove LightWeightOrganization type (#28838)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

evanpurkhiser committed:
ref(js): Simplify performance routes (#28869)

robinrendle committed:
fix(ui): LayoutTitle is now a h1 (#28876)

vuluongj20 committed:
feat(ui): Use tabular figures in tables and charts (#28871)

* Use tabular figures in tables and charts

The CSS style `font-variant-numeric: tabular-nums;` turns on tabular figures - monospaced substitutes for numeric characters in Rubik, ideal for tables and charts. This commits adds the `tabular-nums` CSS style to components across the app.

wmak committed:
ref(discover): Add a deprecated list to discover (#28830)

- This adds a list of known deprecated fields to discover so we can mark
  them as such in the UI
- Culprit is deprecated based on https://github.com/getsentry/relay/pull/1082

wmak committed:
feat(discover): Add referrers to the events-stats endpoint (#28848)

* feat(discover): Add referrers to the events-stats endpoint

- This adds an allow list of referrers to the events-stats endpoint so
  we can differentiate the performance between the different parts of
  the sentry product

wmak committed:
fix(discover): Allow invalid rollups (#28778)

* fix(discover): Allow invalid rollups

- Currently the event-stats endpoint 400s with an error message when a
  user requests an invalid interval
  - But this is happening frequently because the alerts page links to
    discover with a custom interval
  - At which points users try to change the statsPeriod and hit this
    error
- This updates the endpoint to ignore these errors and instead use the
  default for the interval instead
  - I think longer term we should instead change this interval param to
    take "low" to "high" instead and we calculate a precise interval
    based on the period picked
  - But for now this should be a suitable fix to improve the user
    experience

wedamija committed:
fix(semver): Fix bug when negating a semver build filter (#28882)

Using a negation on semver builds would error for any issue/discover searches.

Fixes SENTRY-S4G

evanpurkhiser committed:
style(js): Spacing between conditions (#28874)

joshuarli committed:
chore: update cbtemulator image source (#28872)

priscilawebdev committed:
ref(test): Conver avatar.spec to RTL (#28852)

kamilogorek committed:
feat(js): Handle in_app detection for webpack namespace output (#28859)

matejminar committed:
feat(ui): Do not show event type filter on sessions alert page (#28855)

matejminar committed:
feat(ui): Format alert session chart percentage (#28856)

matejminar committed:
fix(ui): Fix alert threshold validation (#28857)

jjbayer committed:
fix(metrics): Remove UseCase from indexer calls (#28890)

The UseCase arg was removed in #28431.

untitaker committed:
ref(releasehealth): Implement get_release_sessions_time_bounds in metrics backend [INGEST-249] (#28824)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

untitaker committed:
fix(metrics): Tag reprocessing in time_to_process metric (#28891)

untitaker committed:
fix(typing): Fix typing in master (#28893)

loewenheim committed:
feat: Add metric for the rate of symbolicate_event requests [NATIVE-185] (#28565)

ahmedetefy committed:
ref(sessions): Remove unused get_adjacent_releases_based_on_adoption (#28860)

billyvg committed:
build(ui): Allow usage of env var for Sentry Release in frontend (#28878)

This allows the use of an environment variable for the release version when configuring the Sentry SDK. This will allow the frontend release version to diverge from the backend response.

loewenheim committed:
feat: Add metric for durations of symbolicate_event requests [NATIVE-185] (#28791)

priscilawebdev committed:
fix(breadcrumbs): Fix missing html element in the grid (#28900)

untitaker committed:
fix(dependencies): Add typing_extensions as dependency (#28894)

edwardgou-sentry committed:
fix(discover): only disable y axis checkbox if at maximum selected

Updates the yaxis dropdown to not disable entire rows if maximum options are selected, only disable checkboxes

markstory committed:
ref(ui) Update org create to new form library (#28773)

Replace the deprecated form control usage with settings forms.

priscilawebdev committed:
fix(interface-breadcrumb): Fix missing html element for grid (#28904)

robinrendle committed:
fix(ui): made event issue look like a link (#28880)

* fix(ui): made event issue look like a link

* fixed link color in issue details page

AniketDas-Tekky committed:
ref(api-logging): Added user agent to api logging (#28881)

Last metric to be added for now. Track the user agent of the caller

edwardgou-sentry committed:
fix(dashboards): skip renderTag when data.value is undefined

Fixes errors thrown when data.value is undefined for widget fields

maxiuyuan committed:
feat(idp-migrations): one-time account validation keys (#28732)

evanpurkhiser committed:
style(routes): Organize frontend routes.tsx (#28875)

evanpurkhiser committed:
ref(js): Convert DashboardsV2Container to a functional component (#28873)

Zylphrex committed:
fix(ui): Trace navigator nodes with errors misaligned (#28903)

When there is an icon within the node, the node needs to be shifted down so that
it aligns with the other nodes.

wmak committed:
feat(discover): allow up to 10 top events (#28862)

- This updates the top events endpoint so it allows up to 10 top events
  instead of the current limit of 5
- Also updated the test so its more dynamic instead of a hardcoded 50

wmak committed:
ref(discover): Update timeout message to include transaction (#28883)

- This is cause transaction filters can significantly improve
  the performance of discover queries, adding it to the message to help
  users out

billyvg commented on .storybook/preview.tsx:
```diff
@@ -126,6 +126,8 @@ addParameters({
     storySort: {
       order: [
+        'Overview',
+        'Updates',
```
Is `Updates` going to be like a changelog? I might put that first (and maybe even rename to `changelog`?. Also what is `Overview`? Is it more of a "Getting Started"?

billyvg commented on docs-ui/stories/overview.stories.mdx:
```diff
@@ -1,6 +1,6 @@
-import StylesVariantsComponets from '../../images/overview-styles-variants-components.svg';
+import StylesVariantsComponets from '../images/overview-styles-variants-components.svg';
```
I assume this image is specific to overview? I would structure it so that `overview` is its own directory and stick the image there too.

billyvg reviewed (approved):
No blockers, just suggestions.

davidenwang committed:
feat(ui): Teams render props component (#28909)

This is the team equivalent of the <Projects> utility. It is based off the useTeams hook

Zylphrex committed:
feat(suspect-spans): Add group hashes to spans (#28184)

Span descriptions are unbounded free-form strings. This makes it difficult to
store them in snuba for grouping purposes. This change introduces a group hash
to get a fixed-length string that can be used to classify spans in the same
group. The first iteration will use a naive hash of the span description with
additional processing in upcoming changes.

taylangocmen committed:
feat(comparison_alerts): Change alert condition selector for issue alerts (#28733)

* feat(comparison_alerts): Update issue alert api to support comparison values

This allows issue alerts with comparison values to be created, and returns these new fields back
from the appropriate endpoints.

* remove new fields from form fields

* feat(alerts): Change alert condition selector for issue alerts

* move change alert node manip to a file

* pick correct item

* refactor

* style(lint): Auto commit lint changes

* style(lint): Auto commit lint changes

* addressing comments

* style(lint): Auto commit lint changes

* revert type

* fix initial value for comparisonInterval

Co-authored-by: Dan Fuller <dfuller@sentry.io>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

evanpurkhiser committed:
ref(routes): Fix legacy /:orgId redirects (#28918)

evanpurkhiser committed:
ref(js): Convert OrganizationDetails* to functional components (#28921)

evanpurkhiser committed:
ref(js): Remove keydown usage in PlatformPicker (#28932)

evanpurkhiser committed:
ref(hooks): Remove routes:organization-root hook (#28913)

evanpurkhiser committed:
ref(js): Convert AppRoot to a functional component (#28925)

evanpurkhiser committed:
ref(js): Wrap all route names in translations (#28934)

evanpurkhiser committed:
ref(js): Remove unused location context (#28928)

priscilawebdev committed:
feat(interface-breadcrumb): Add breadcrumb eventlink (#28889)

priscilawebdev committed:
fix(interface-breadcrumb): Disable linked events (#28939)

matejminar committed:
feat(ui): Scale the alert creation chart (#28896)

loewenheim committed:
fix: Properly export increment_project_duration_counter (#28941)

RaduW committed:
ref(releasehealth): Implement get_changed_project_release_model_adoptions on metrics backend [INGEST-249] (#28895)

Port get_changed_project_release_model_adoptions from sentry.snuba.sessions to metrics.

k-fish committed:
ref(sdk): Testing default idleTimeout (#28868)

The default idleTimeout for our sdk is 1000ms, many months back we increased the number from the default to increase our collection of LCP, but we don't have any hard numbers on the impact. This will help us test our LCP collection, although it could have some impact on trends.

relaxolotl committed:
ref: Bump symbolic version (#28911)

Includes a hotfix for getsentry/sentry-dart#591, by 
making parsing more lenient and resilient against 
malformed ELFs.

armenzg committed:
feat(ci): python-deps workflow to install specific packages (#28901)

This reduces the number of packages that get installed by brew and it prevents upgrading packages.

k-fish committed:
feat(perf): Generic landing widgets (all transactions tab) (#28861)

* feat(perf): Generic landing widgets

This adds some initial generic landing widgets for the 'all transactions' tab of the new performance landing pages. It introduces a widget container component, which has widget definitions. These definitions will use a specialization of the genericPerformanceWidget (eg. singleFieldAreaWidget). Widgets are split into their queries and visualizations (such as charts and tables), so that data can be fetched or batched later independently of rendering ththe representation. For queries, all queries go through a QueryHandler which sets data from the queries onto a widgetData object which is available throughout the GenericPerformanceWidget. The query handler is currently designed to handle our waterfall style component apis as well as run transform functions to normalize the results into something the generic performance widget can use to switch between widget states.

The query-handler should be somewhat temporary until we have a path to move away from waterfall queries or we decide to remake all our queries functionally.

There is some memoization currently required in the specializations for the performance widget which should be able to be removed once I finish making the remaining widgets and can see what the final component api looks like.

evanh committed:
ref(snql) Add a migration to recreate subscriptions in SnQL (#28835)

This migration will recreate all the existing subscriptions in Snuba. New
subscriptions are created with SnQL, so this effectively rebuilds all existing
subscriptions with SnQL.

This is marked as dangerous because it has already been run manually in prod and
there's no point in running it again. This has been added so that we can
deprecate the Snuba legacy API in later releases.

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

mgaeta committed:
ref(integrations): Move functions to Model Managers (#28885)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

leeandher committed:
test(alert-rules): Add a barebones spec test for the RuleNode component (#28834)

iProgramStuff committed:
feat(workflow): Team Alerts Triggered API (#28816)

scttcper committed:
test(ui): Convert clipboard tooltip component to RTL (#28930)

scttcper committed:
fix(ui): Force refresh projects/teams on project creation (#28886)

mgaeta committed:
fix(integrations): Integration ID TypeError (#28916)

k-fish committed:
Revert "ref(sdk): Testing default idleTimeout (#28868)" (#28949)

This reverts commit 443da5fee1b28354d17757c7dc1c35f8a7a3eed8.

beezz committed:
ref(db): Explicitly use replica for platform stats task (#28897)

* ref(db): Explicitly use replica for platform stats task

* ref(db): Add `using_replica` to the base model manager

* Move the `using_replica` to a BaseQuerySet

* Update sentry BaseManager to reflect methods from sentry BaseQuerySet

markstory committed:
ref(deletion) Remove unused deletion tasks (#28799)

Remove custom deletion tasks that are no longer in use. Deletions use ScheduledDeletion instead now.

markstory committed:
feat(docs) Add docs for deletions subsystem (#28915)

I thought these docs were a little too low-level and specific to be in
develop.sentry.io and decided to expand the existing module docstring
instead.

iProgramStuff committed:
feat(workflow): GroupHistory model (#28691)

This model will store various states of history for groups. It differs from GroupResolution because it stores other status', and is intended to be easier to quickly query than Activity.

mgaeta committed:
ref(integrations): f-strings (#28887)

nikhars committed:
feat(post-process-forwarder) Functioning batching batching_kafka_consumer (#28801)

This PR integrates the different components to build a functioning
batching kafka consumer for post process forwarder.

The older streaming consumer is removed and all the business logic
of processing has been moved to PostProcessWorker class.

mgaeta committed:
fix(integrations): User is None (#28955)

mgaeta committed:
ref(slack): Clean up tests (#28846)

scttcper committed:
feat(workflow): Team alerts triggered graph (#28845)

vuluongj20 committed:
feat(ui): Move to Roboto Mono (#28956)

* Move to Roboto Mono

Moving to Roboto Mono (from IBM Plex) due to its higher visual harmony (in terms of weight and letter design) with Rubik.

davidenwang committed:
feat(team-store): Support user teams state in TeamStore (#28957)

Augment the team store with state relating to whether or not the user's teams are loaded. This will be useful to avoid continually refetching the user's teams.

evanpurkhiser committed:
ref(js): Remove context usage from monitors (#28935)

mgaeta committed:
fix(slack): Unlink Identity (#28811)

markstory committed:
fix(deletion) Fix snuba query deletion problem (#28953)

Organization's that use AlertRules are getting stuck in the deletion
queues because the `post_delete` signal is failing. The chain of events
is:

- Org deletion starts
- Environment is deleted.
- SnubaQuery has a foreign key to Environment, so django deletes
  SnubaQuery
- AlertRule has a foreign key to SnubaQuery, so django deletes it.
- AlertRule post_deletion is fired, and the snuba_query is gone causing
  the read to fail

Reordering the relation deletion so AlertRules are removed first breaks
this chain and ensures that alert rules are removed before their snuba
queries are removed.

Fixes SENTRY-SAB

scttcper committed:
test(ui): Convert checkbox fancy to RTL (#28959)

evanpurkhiser committed:
ref(js): Replace react-keydown with react-hotkeys-hook (#28933)

evanpurkhiser committed:
ref(js): Fix link to projects install from settings (#28969)

evanpurkhiser committed:
ref(js): Convert App to a functional component (#28936)

vuluongj20 committed:
Restructure docs-ui

Based on Billy's suggestions in #28788, restructuring docs-ui in the following ways:

 - Move .stories.* files with dependencies (images, special components) into their own directories

 - Rename 'Overview' to 'Getting Started', and 'Updates' to 'Changelog'

getsantry[bot] committed:
style(lint): Auto commit lint changes

mitsuhiko committed:
build(sdk): Bump Python SDK to 1.4.3 (#28984)

matejminar committed:
fix(ui): Refetch alert chart total when project changes (#28940)

matejminar committed:
feat(ui): Add 30 minutes window to crash rate alerts (#28938)

ahmedetefy committed:
fix(semver): Set semver cols on pre_save (#28946)

Creates a pre_save signal that parses release
version into semver on creation

evanpurkhiser committed:
ref(ts): Add some missing typing (#28983)

evanpurkhiser committed:
ref(js): Minor cleanup to InstallWizard (#28968)

evanpurkhiser committed:
perf(js): Avoid rebuilding routes w/ memoization (#28975)

matejminar committed:
fix(ui): Fix alerts supported search tags (#28985)

ahmedetefy committed:
ref(tasks): Safeguard post_process_group with try/except (#28986)

RaduW committed:
ref(releasehealth): Implement get_oldest_health_data_for_releases on metrics backend [INGEST-249] (#28943)

* implement get_oldest_health_data_for_releases in metrics backend

vuluongj20 commented:
Closing due to bad pull from master that somehow made me co-author of all the upstream commits.
